code/modules/clothing/glasses/glasses.dm

```diff
@@ -160,6 +160,89 @@
 	prescription_upgradable = FALSE
 	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE //don't render darkness while wearing these
 
+// Has an action button when equipped that shows information about healing chemicals and how to create them
+/obj/item/clothing/glasses/medchem
+	name = "medical chemistry goggles"
+	desc = "A pair of goggles detecting and analyzing reagents. With an inbuilt holoprojector, they are able to display information about various healing chemicals."
+	icon_state = "purple"	// todo
+	item_state = "glasses"	// todo2
+	origin_tech = "biotech=4;materials=4"
+	prescription_upgradable = TRUE
+	scan_reagents = TRUE
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, RAD = 0, FIRE = 200, ACID = INFINITY)
+	sprite_sheets = list(
+		"Vox" = 'icons/mob/clothing/species/vox/eyes.dmi',
+		"Grey" = 'icons/mob/clothing/species/grey/eyes.dmi',
+		"Drask" = 'icons/mob/clothing/species/drask/eyes.dmi',
+		"Kidan" = 'icons/mob/clothing/species/kidan/eyes.dmi')
+	actions_types = list(/datum/action/item_action/toggle_medchem_recipes)
+
+	/// The list of reagents that it should show information about, use /datum/reagent's "id" here
+	var/list/medical_reagents = list("antihol", "atropine", "calomel", "charcoal", "cryoxadone", "degreaser", "diphenhydramine", "ephedrine", "epinephrine", "haloperidol", "heparin", "hydrocodone", "lazarus_reagent", "liquid_solder", "mannitol", "mitocholide", "mutadone", "oculine", "pen_acid", "perfluorodecalin", "potass_iodide", "rezadone", "salbutamol", "salglu_solution",  "sal_acid", "sanguine_reagent", "silver_sulfadiazine", "spaceacillin", "styptic_powder", "synthflesh",  "teporone")
+	/// List of medicine we will pass over to tgui
+	var/list/medicine_list = list()
+	/// List of components we will pass over to tgui
```

Comment:
I recommend renaming components to something like ingredients, because we already have something named components in code

Reply:
Code-wise I'd agree, but these things show up as "components" on tgui and are called "components" on the chem wiki page, so for consistency, the list is called `component_list`
unless we got a better word for it, in which case I'll rename it at both places